vm/actor/src/builtin/multisig/mod.rs

```diff
@@ -0,0 +1,425 @@
+            if st.is_signer(&params.to) {
+                rt.abort(
+                    ExitCode::ErrIllegalArgument,
+                    "Party already present".to_owned(),
+                )
+            }
+
+            // Remove signer from state
+            st.signers.retain(|s| s != &params.from);
+
+            // Add new signer
+            st.signers.push(params.to);
+        });
+    }
+
+    /// Multisig actor function to change number of approvals needed
+    pub fn change_num_approvals_threshold<BS, RT>(
+        rt: &RT,
+        params: ChangeNumApprovalsThresholdParams,
+    ) where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_is(std::iter::once(rt.message().to()));
+
+        rt.transaction::<State, _, _>(|st| {
+            // Check if valid threshold value
+            if params.new_threshold <= 0 || params.new_threshold as usize > st.signers.len() {
+                rt.abort(
+                    ExitCode::ErrIllegalArgument,
+                    "New threshold value not supported".to_owned(),
+                );
+            }
+
+            // Update threshold on state
+            st.num_approvals_threshold = params.new_threshold
+        });
+    }
+
+    fn approve_transaction<BS, RT>(rt: &RT, tx_id: TxnID)
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        // Approval transaction
+        let (tx, threshold_met): (Transaction, bool) = rt.transaction::<State, _, _>(|st| {
+            let mut txn = match st.get_pending_transaction(rt.store(), tx_id) {
+                Ok(t) => t,
+                Err(e) => {
+                    rt.abort(
+                        ExitCode::ErrIllegalState,
+                        format!("Failed to get transaction for approval: {}", e),
+                    );
+                    unreachable!()
+                }
+            };
+
+            // abort duplicate approval
+            for previous_approver in &txn.approved {
+                if previous_approver == rt.message().from() {
+                    rt.abort(
+                        ExitCode::ErrIllegalState,
+                        "Already approved this message".to_owned(),
+                    )
+                }
+            }
+
+            // update approved on the transaction
+            txn.approved.push(rt.message().from().clone());
+
+            if let Err(e) = st.put_pending_transaction(rt.store(), tx_id, txn.clone()) {
+                rt.abort(
+                    ExitCode::ErrIllegalState,
+                    format!("Failed to put transaction for approval: {}", e),
+                );
+            }
+
+            // Check if number approvals is met
+            if txn.approved.len() >= st.num_approvals_threshold as usize {
+                if let Err(e) =
+                    st.check_available(rt.current_balance(), txn.value.clone(), rt.curr_epoch())
+                {
+                    // Ensure sufficient funds
```

Comment:
should this comment exist on line 340? It seems out of place where it sits currently.